identity/backend/src/main/java/io/camunda/identity/usermanagement/service/UserService.java

```diff
@@ -70,23 +80,30 @@ public List<CamundaUser> findAllUsers() {
     return userRepository.loadUsers();
   }
 
-  public CamundaUser updateUser(final String username, final CamundaUserWithPassword user) {
+  @Transactional
+  public CamundaUser updateUser(final Integer id, final CamundaUserWithPassword user) {
     try {
-      if (!username.equals(user.user().username())) {
+      if (!Objects.equals(id, user.user().id())) {
+        throw new RuntimeException("user.notFound");
+      }
+      final CamundaUser existingUser = userRepository.loadUserById(id);
+      if (existingUser == null || !existingUser.username().equals(user.user().username())) {
         throw new RuntimeException("user.notFound");
       }
 
-      final UserDetails existingUser = userDetailsManager.loadUserByUsername(username);
+      final UserDetails existingUserDetail =
+          userDetailsManager.loadUserByUsername(existingUser.username());
 
       final UserDetails userDetails =
-          org.springframework.security.core.userdetails.User.withUsername(username)
+          org.springframework.security.core.userdetails.User.withUsername(existingUser.username())
```

Comment:
Any chance to not have to use the full path here? Always a bit difficult for readability to me. WDYT?